handler/enqueue_annotation.go

```diff
@@ -112,16 +171,30 @@ func parseNamespacedName(namespacedNameString string) types.NamespacedName {
 	return types.NamespacedName{}
 }
 
-// SetOwnerAnnotation sets annotations for dependent resources that needs to be watched by namespaced Owners.
-func SetOwnerAnnotation(u *unstructured.Unstructured, owner *unstructured.Unstructured) {
-	a := u.GetAnnotations()
-	if a == nil {
-		a = map[string]string{}
+// SetWatchOwnerAnnotation is a helper method to add watch annotation-based with the owner NamespacedName and
+// schema.GroupKind to the object provided. This allows you to declare the watch annotations of an owner to an object.
+// If a annotation to the same object already exists, it'll be overwritten with the newly provided version.
+func SetOwnerAnnotation(owner, object metav1.Object, ownerGK schema.GroupKind) error {
```

Comment:
See that it is a helper. so, we need to update the places in SDK that are using the annotation.